### ISSUE TITLE
AWS Workspaces - Zero Results even though there are Workspaces

### ISSUE THREAD

**Issue by jonabend-uptake** [turbot/steampipe-plugin-aws#920]
**Describe the bug**
When I run query for workspaces, steampipe returns empty table even though there are multiple Workspaces in the AWS Account.
```
18:16 $ steampipe query
Welcome to Steampipe v0.12.2
For more information, type .help
> .cache clear
> select
  workspace_id
from
  aws_workspaces_workspace
+--------------+
| workspace_id |
+--------------+
+--------------+
```

Output from AWS CLI from same account with same credentials. (Output truncated and obfuscated)
```
18:10 $ aws workspaces describe-workspaces --query 'Workspaces[].WorkspaceId'
[
    "ws-wr......",
    "ws-bc......",
    "ws-62......",
    "ws-d1......",
    "ws-97......"
]
```

**Steampipe version (`steampipe -v`)**
```
steampipe version 0.12.2
```

**Plugin version (`steampipe plugin list`)**
```
+---------------------------------------------+---------+-------------+
| Name                                        | Version | Connections |
+---------------------------------------------+---------+-------------+
| hub.steampipe.io/plugins/turbot/aws@latest  | 0.49.0  | aws         |
| hub.steampipe.io/plugins/turbot/jira@latest | 0.1.0   | jira        |
+---------------------------------------------+---------+-------------+
```

**To reproduce**

1. In AWS Account where Workspaces exist, run query
`select workspace_id from aws_workspaces_workspace`

**Expected behavior**
Output should list the Workspaces IDs for the existing Workspaces

**Comment by cbruno10** [comment]
Hey @jonabend-uptake , can you please share what region the expected workspaces are in? I'm thinking it may be something related to how we try and handle unsupported regions (https://github.com/turbot/steampipe-plugin-aws/blob/main/aws/table_aws_workspaces_workspace.go#L158-L164), which may be causing us to filter out results from regions that actually are supported.

**Comment by jonabend-uptake** [comment]
@cbruno10 the AWS Region is: `us-east-1`

Please let me know if there is any additional information I can provide or troubleshooting steps to perform.  

I have reviewed the logs in `~/.steampipe/logs` and set `STEAMPIPE_LOG=trace`.  I did not see details on the call to the AWS endpoints in either.

**Comment by c0d3r-arnab** [comment]
Hi @jonabend-uptake, while testing we found out that it is working fine on our end -
Configuration in .spc file with all region-
```
connection "aws" {
access_key = "*****"
secret_key = "*****"
regions = ["*"]    // All regions
plugin    = "aws"               
}

> select workspace_id from aws_workspaces_workspace
+--------------+
| workspace_id |
+--------------+
| ws-2spwmjtl8 |
+--------------+
```
Configuration in .spc file with us-east-1-
```
connection "aws" {
access_key = "*****"
secret_key = "*****"
regions = ["us-east-1"] 
plugin    = "aws"               
}

> select workspace_id from aws_workspaces_workspace
+--------------+
| workspace_id |
+--------------+
| ws-2spwmjtl8 |
+--------------+
```
Could you please check if the `regions` parameter is properly configured in aws.spc file?

**Comment by jonabend-uptake** [comment]
@cbruno10 I updated my aws.spc to include the regions
`regions = ["us-east-1"]`
Now the query for aws_workspaces_workspace is returning the expected results.

Up to this point, I have exported AWS_PROFILE and AWS_REGION before using Steampipe.

I have done additional testing by now commenting out the `regions` line in my aws.spc file, and the Steampipe query DOES return the aws_workspaces_workspace results as expected.

I am now, no longer able to reproduce the issue.

Thank you for taking the time to research this and provide guidance.  I will close this issue as Steampipe is working properly.
